src/benchmark/operators/sort_benchmark.cpp

```diff
@@ -1,22 +1,181 @@
+ public:
+  void SetUp(benchmark::State& st) override {
+    SortPicoBenchmark::SetUp(st);
+    MakeReferenceTable();
+  }
+};
+
+class SortReferenceSmallBenchmark : public SortSmallBenchmark {
+ public:
+  void SetUp(benchmark::State& st) override {
+    SortSmallBenchmark::SetUp(st);
+    MakeReferenceTable();
+  }
+};
+
+class SortReferenceBenchmark : public SortBenchmark {
+ public:
+  void SetUp(benchmark::State& st) override {
+    SortBenchmark::SetUp(st);
+    MakeReferenceTable();
+  }
+};
+
+class SortReferenceLargeBenchmark : public SortLargeBenchmark {
+ public:
+  void SetUp(benchmark::State& st) override {
+    SortLargeBenchmark::SetUp(st);
+    MakeReferenceTable();
+  }
+};
+
+class SortStringSmallBenchmark : public SortBenchmark {
+ public:
+  void SetUp(benchmark::State& st) override {
+    GenerateCustomTable(ChunkOffset{2'000}, size_t{4'000}, DataType::String);
+  }
+};
+
+class SortStringBenchmark : public SortBenchmark {
+ public:
+  void SetUp(benchmark::State& st) override {
+    GenerateCustomTable(ChunkOffset{2'000}, size_t{40'000}, DataType::String);
+  }
+};
+
+class SortStringLargeBenchmark : public SortBenchmark {
+ public:
+  void SetUp(benchmark::State& st) override {
+    GenerateCustomTable(ChunkOffset{2'000}, size_t{400'000}, DataType::String);
+  }
+};
+
+class SortNullBenchmark : public SortBenchmark {
+ public:
+  void SetUp(benchmark::State& st) override {
+    const auto table_generator = std::make_shared<SyntheticTableGenerator>();
+    const int num_columns = 1;
+    const int max_different_value_a = 40'000;
+    const int max_different_value_b = 20'000;
+    const size_t chunk_size{2'000};
+    const std::vector<DataType> column_data_types = {num_columns, DataType::Int};
+
+    _table_wrapper_a = std::make_shared<TableWrapper>(table_generator->generate_table(
+        {num_columns, {ColumnDataDistribution::make_uniform_config(0.0, max_different_value_a)}}, column_data_types,
+        ChunkOffset{max_different_value_a}, chunk_size,
+        std::vector<SegmentEncodingSpec>(num_columns, {EncodingType::Unencoded}), std::nullopt, UseMvcc::No));
+    _table_wrapper_a->execute();
+
+    _table_wrapper_b = std::make_shared<TableWrapper>(table_generator->generate_table(
+        {num_columns, {ColumnDataDistribution::make_uniform_config(0.0, max_different_value_b)}}, column_data_types,
+        ChunkOffset{max_different_value_b}, chunk_size,
+        std::vector<SegmentEncodingSpec>(num_columns, {EncodingType::Unencoded}), std::nullopt, UseMvcc::No));
+    _table_wrapper_b->execute();
+
+    auto join = std::make_shared<JoinHash>(
+        _table_wrapper_a, _table_wrapper_b, JoinMode::Left,
+        OperatorJoinPredicate{ColumnIDPair(ColumnID{0}, ColumnID{0}), PredicateCondition::Equals});
+    join->execute();
+
+
+    // TODO(sorting-group): Find a more reliable way to produce an exct number of NULL values. Right now this produces
+    //  ~20% NULL Values of the bigger table
```

Comment:
IMO, the cleanest way would be to extend SyntheticTableGenerator to allow for a "null ratio" in the produced data.

Reply:
Resolved in 190d705a9b7e052f12796786604a79ea90bf5fe4